### PR TITLE
Swap matter.js physics for wasm rapier.js

### PR DESCRIPTION
**Pull request by LeoSipowicz**
Changed the ball drop physics from using the slower matter.js library to rapier.js for increased performance and functionality. The physics are now rendered using pixi.js and there is a new mouse interactivity feature. A demo video of what the new physics looks like is embedded below.
Rapier_phys_demo

A new setup-jest.js file was also needed with text textEncoder/Decoder globals in order to workaround known jest issue with a jsdom enviorment. More info about the bug [here](https://github.com/inrupt/solid-client-authn-js/issues/1676).

### CODE REVIEW

File: app/javascript/components/widgets/numbers/panel.jsx [hunk 1]
```diff
@@ -7,8 +7,9 @@ import styled from 'styled-components';
 import pluralize from 'pluralize';
 import { getStartOfWeek } from '@lib/datetime';
 import { LoadingMessage, DisconnectedMessage } from '@messages/message';
-import { colors, weights, fontSizes, fonts, spacing } from '@lib/theme';
-import FallingBlocks from '@numbers/falling-blocks';
+import { colors, weights, fontSizes, fonts, spacing 
+} from '@lib/theme';
```
Comment (aaddaamm) [inline app/javascript/components/widgets/numbers/panel.jsx:11]: CI is upset that this is now on a newline. lets remove this diff to appease it.

**Comment by aaddaamm** [comment]
lots of lint issues in the new files. running `yarn run lint` locally will help with figuring out the problems, you can likely solve most of them by using `--fix` as a parameter like the build output suggests.

**Comment by aaddaamm** [comment]
Nice work on the commits

File: app/javascript/components/widgets/numbers/rapierPixi.js [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    shape.xSize = sizeX;
+    shape.ySize = sizeY;
+    ColliderMap.set(collider.handle, shape);
+  }
+
+  //Render each object in ColliderMap in PixiJS graphics
+  function render(world, ColliderMap) {
+    let cntr = 0;
+    ColliderMap.forEach((el) => {
+      3;
```
Comment (cbarber) [inline app/javascript/components/widgets/numbers/rapierPixi.js:163]: What does this do?

File: app/javascript/components/widgets/numbers/rapierPixi.js [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+  let rightWallBodyDesc = RAPIER.RigidBodyDesc.fixed().setTranslation(
+    windowSizeX - border,
+    0,
+  );
+  let rightWallBody = world.createRigidBody(rightWallBodyDesc);
+  let rightWallColliderDesc = RAPIER.ColliderDesc.cuboid(10, windowSizeY);
+  world.createCollider(rightWallColliderDesc, rightWallBody);
+
+  let num = 1;
+  let numy = githubPullCount + githubCommitCount + slackMessageCount;
```
Comment (cbarber) [inline app/javascript/components/widgets/numbers/rapierPixi.js:51]: Maybe some specific names? `totalCount`?

File: app/javascript/components/widgets/numbers/rapierPixi.js [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+  let shift = rad * 1.5 + rad;
+  let centerx = shift * (num / 2) - 400;
+  let centery = shift / 2.0;
+
+  let i, j;
+
+  //Rapier falling cubes (dynamic)
+  for (j = 0; j < numy; ++j) {
+    for (i = 0; i < num; ++i) {
```
Comment (cbarber) [inline app/javascript/components/widgets/numbers/rapierPixi.js:63]: `num` is always 1. Is this loop necessary?

File: app/javascript/components/widgets/numbers/rapierPixi.js [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+        } else if (rand == 1 && githubCommitCount > 0) {
+          spotFound = true;
+          githubCommitCount--;
+          img = githubCommitIcon;
+        } else if (rand == 2 && slackMessageCount > 0) {
+          spotFound = true;
+          slackMessageCount--;
+          img = slackMessageIcon;
+        }
+      }
```
Comment (cbarber) [inline app/javascript/components/widgets/numbers/rapierPixi.js:89]: It took me a little bit to understand what this loop was doing. Maybe extract a method and prevent a potential infinite loop?

File: package.json [hunk 6]
```diff
@@ -136,12 +136,9 @@
   },
   "jest": {
     "verbose": true,
-<<<<<<< HEAD
```
Comment (chewnoill) [inline package.json]: Conflict markers got squashed into the wrong place 🕵️‍♂️

**Review by cbarber — APPROVED** [review]
:rocket: This is awesome!

**Pull request merged**